x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -268,13 +207,25 @@ export const StepDetailsForm: FC<Props> = React.memo(
       .filter((f) => f.type === KBN_FIELD_TYPES.DATE)
       .map((f) => f.name)
       .sort();
+
+    // Continuous Mode
     const isContinuousModeAvailable = dateFieldNames.length > 0;
     const [continuousModeDateField, setContinuousModeDateField] = useState(
       isContinuousModeAvailable ? dateFieldNames[0] : ''
     );
     const [continuousModeDelay, setContinuousModeDelay] = useState(defaults.continuousModeDelay);
     const isContinuousModeDelayValid = continuousModeDelayValidator(continuousModeDelay);
 
+    // Retention Policy
+    const isRetentionPolicyAvailable = dateFieldNames.length > 0;
+    const [retentionPolicyDateField, setRetentionPolicyDateField] = useState(
+      isRetentionPolicyAvailable ? dateFieldNames[0] : ''
+    );
+    const [retentionPolicyMaxAge, setRetentionPolicyMaxAge] = useState(
+      defaults.retentionPolicyMaxAge
+    );
+    const isRetentionPolicyMaxAgeValid = continuousModeDelayValidator(continuousModeDelay);
```

Comment:
Copy and paste error - this needs to check against `isRetentionPolicyMaxAgeValid`

Reply:
Fixed in 9f652a1 / 20e47d2.